ios/Sources/PrivacyScreenPlugin/PrivacyScreenPlugin.swift

```diff
@@ -0,0 +1,145 @@
+            default:
+                blurEffect = nil
+            }
+        }
+
+        isEnabled = true
+        call.resolve(["success": true])
+    }
+
+    @objc func disable(_ call: CAPPluginCall) {
+        isEnabled = false
+        unobscureScreen()
+        call.resolve(["success": true])
+    }
+
+    @objc func isEnabled(_ call: CAPPluginCall) {
+        call.resolve(["enabled": isEnabled])
+    }
+
+    private func obscureScreen() {
+        guard let window = getKeyWindow(),
+              var topViewController = window.rootViewController else {
+            return
+        }
+
+        while let presentedVC = topViewController.presentedViewController {
+            topViewController = presentedVC
+        }
+
+        if screenProtectionViewController == nil {
+            let protectionVC = createProtectionViewController()
+            topViewController.present(protectionVC, animated: false)
+            screenProtectionViewController = protectionVC
+        }
+    }
+
+    private func unobscureScreen() {
+        screenProtectionViewController?.dismiss(animated: false)
+        screenProtectionViewController = nil
+    }
+
+    private func createProtectionViewController() -> UIViewController {
+        let viewController = UIViewController()
+        viewController.modalPresentationStyle = .overFullScreen
+        viewController.view.isUserInteractionEnabled = false
+
+        if let blurEffect = blurEffect {
+            let blurEffectView = UIVisualEffectView(effect: UIBlurEffect(style: blurEffect))
+            blurEffectView.frame = UIScreen.main.bounds
+            blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            viewController.view.addSubview(blurEffectView)
+        } else {
+            let contentView = UIView(frame: UIScreen.main.bounds)
+            contentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+
+            if let launchImage = UIImage(named: "LaunchImage") ?? UIImage(named: "Splash") {
+                let imageView = UIImageView(image: launchImage)
+                imageView.frame = contentView.bounds
+                imageView.contentMode = .scaleAspectFill
+                imageView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+                contentView.addSubview(imageView)
+            } else {
+                if let launchStoryboard = Bundle.main.object(forInfoDictionaryKey: "UILaunchStoryboardName") as? String,
+                   let launchVC = UIStoryboard(name: launchStoryboard, bundle: nil).instantiateInitialViewController() {
+                    contentView.backgroundColor = launchVC.view.backgroundColor
+                } else {
+                    contentView.backgroundColor = .white
+                }
+            }
+
+            viewController.view = contentView
+        }
+
+        return viewController
+    }
+
+    private func getKeyWindow() -> UIWindow? {
+        if #available(iOS 13.0, *) {
```

Comment:
Looking at the podspec, iOS 14 is the minimum version we support. So what's the point of this if statement?